src/environment.cpp

```diff
@@ -821,18 +821,22 @@ class ABMHandler
 						k != ids.end(); ++k)
 				{
 					content_t c = *k;
-					std::map<content_t, std::vector<ActiveABM> >::iterator j;
-					j = m_aabms.find(c);
-					if(j == m_aabms.end()){
-						std::vector<ActiveABM> aabmlist;
-						m_aabms[c] = aabmlist;
-						j = m_aabms.find(c);
-					}
-					j->second.push_back(aabm);
+					if (c >= m_aabms.size())
+						m_aabms.resize(c + 256, (std::vector<ActiveABM> *) NULL);
```

Comment:
It's vital to the ::apply method that these are initialized to NULL. While it might not be necessary, it's nice to leave the initialization in to indicate such.

The cast is obviously unnecessary and annoying.